fontforge/fvfonts.c

```diff
@@ -1267,6 +1267,86 @@ return;
+bool InterpolationSanity(SplineSet* base, SplineSet *other, int order2, char* prefix) {
+    SplineSet *bss=NULL, *oss=NULL;
+    int bmcc = 0, omcc = 0;
+    bool ret = true;
+
+    if (base == NULL || other == NULL) return false;
+
+    // Count contours...
+    for ( bss = base; bss != NULL; bss=bss->next ) bmcc++;
+    for ( oss = other; oss != NULL; oss=oss->next ) omcc++;
+
+    // An empty glyph is compatible.
+    if (bmcc == 0 && omcc == 0) return true;
+
+    if (prefix == NULL) prefix = "";
+    else prefix = smprintf(_("Glyph %s: "), prefix);
+
+    if (bmcc != omcc) {
+        LogError(_("%sIncompatible contour counts: %d vs %d."), prefix, bmcc, omcc);
+        ret = false;
+    }
+
+    SplinePoint* sp;
+    int contour_count = 1;
+    for ( bss = base, oss = other; bss!=NULL; bss=bss->next, oss=oss->next, contour_count++ ) {
+        if (!ret) break;
+        int bicw = SplinePointListIsClockwise(bss), oicw = SplinePointListIsClockwise(oss);
+
+        if (bicw == -1 || oicw == -1)
+            TRACE(_("%sInterpolationSanity: Assuming self-intersecting contour %d to be sane"), prefix, contour_count);
+        else if (bicw != oicw) {
+            LogError(_("%sIncompatible contour directionality (contour %d)"), prefix, contour_count);
+            ret = false;
+        }
+
+        int bpnum = SplineSetNumberPoints(bss, order2);
+        int opnum = SplineSetNumberPoints(oss, order2);
+        if (bpnum != opnum) {
+            LogError(_("%sIncompatible point counts: %d vs %d (countour %d)"), prefix, bpnum, opnum, contour_count);
+            ret = false;
+        }
+
+        int segment_count = 1;
+        for ( SplinePoint *bsp=bss->first, *osp=oss->first; ; segment_count++ ) {
+            bool bnonextpt = bsp->next==NULL;
+            bool ononextpt = osp->next==NULL;
+            if (bnonextpt != ononextpt) {
+                char* missingnextpt = bnonextpt ? _("base") : _("other");
+                LogError(_("%sWe expected another point in the %s spline after segment %d"), prefix, missingnextpt, segment_count);
+                ret = false; break;
+            }
+            bool bcurve = ( !bsp->nonextcp || (!bnonextpt && !bsp->next->to->noprevcp) );
+            bool ocurve = ( !osp->nonextcp || (!ononextpt && !osp->next->to->noprevcp) );
+            if ( bcurve != ocurve ) {
+                char* wrongpttype = bcurve ? _("base") : _("other");
+                LogError(_("%sSegment %d in the %s spline is a curve when it should be a line"), prefix, segment_count, wrongpttype);
+                ret = false; break;
+            }
+            if ( bnonextpt ) break;
+            bsp = bsp->next->to;
+            osp = osp->next->to;
+            if ( bsp==bss->first && osp==oss->first ) break;
+            else if ( bsp==bss->first || osp==oss->first ) {
+                ret = false; break;
+            }
+        }
+    }
+
+    if (prefix[0] != '\0') free(prefix);
```

Comment:
I think it's better in general for the allocating level to do the free -- it knows how the memory was allocated and the lifecycle of the string.

Reply:
Sorry, not sure I understand. This function is the allocating level. (Yes I said I was going to sleep, having trouble with that.)

Reply:
Oh, I see, you're assigning above. I'm not sure I'm thrilled about that in terms of generality but I guess it's fine. Easy enough to refactor later.